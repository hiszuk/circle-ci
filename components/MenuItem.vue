<template>
  <div class="menu-item">
    <a class="menu-item-label" @click="clickMenuItem">
      {{ label }}
    </a>
  </div>
</template>

<script>
export default {
  name: 'MenuItem',
  props: {
    label: {
      type: String,
      default: '',
      required: true
    },
    name: {
      type: String,
      default: '',
      required: true
    }
  },
  methods: {
    clickMenuItem() {
      this.$emit('clickMenuItem', {name: this.name})
    }
  },
}
</script>
