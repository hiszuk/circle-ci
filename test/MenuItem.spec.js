import { shallowMount } from "@vue/test-utils";
import MenuItem from "@/components/MenuItem.vue";

const label = "MenuLabel";
const name = "linkurl";
const factory = () => {
  return shallowMount(MenuItem, {
    propsData: {
      label: label,
      name: name
    }
  });
};

describe("MenuItem.vue", () => {
  describe("プロパティ受け渡し", () => {
    it("リンクボタン表示", () => {
      const wrapper = factory();
      expect(wrapper.vm.$props.label).toBe(label);
      expect(wrapper.vm.$props.name).toBe(name);
      expect(typeof wrapper.vm.$props.label).toBe("string");
      expect(typeof wrapper.vm.$props.name).toBe("string");
      expect(wrapper.vm.$options.props.label.required).toBe(true);
      expect(wrapper.vm.$options.props.name.required).toBe(true);
    });
  });

  describe("template", () => {
    it("snapshot", () => {
      const wrapper = factory();
      // レンダリング結果がスナップショットと同じか
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });

  describe("MenuItemClick", () => {
    it("ボタンクリックでメソッドが起動されるか?", () => {
      const wrapper = factory();

      // DOMを探してクリックする
      wrapper.find(".menu-item-label").trigger("click");
      // mockがコールされたか？
      expect(wrapper.emitted().clickMenuItem[0]).toEqual([{ name: name }]);
    });
  });
});
