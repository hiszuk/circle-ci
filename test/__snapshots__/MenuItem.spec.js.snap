// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem.vue template snapshot 1`] = `
<div
  class="menu-item"
>
  <a
    class="menu-item-label"
  >
    
    MenuLabel
  
  </a>
</div>
`;
